src/lib/Service/ServiceProviderInterface.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\SystemInfo\Service;
+
+/**
+ * @internal
+ */
+interface ServiceProviderInterface
+{
+    public function provide(string $identifier): Service;
```

Comment:
In vast majority of cases the intermediate step can be skipped, and this interface can directly return the service type.
```suggestion
    public function getServiceType(string $identifier): string;
```
If the `Service` access becomes required at any point, an additional method can be introduced to get it. But I consider the fact that there is an object beneath an implementation detail (even if you do provide an interface for it).